common/txmgr/types/tx.go

```diff
@@ -66,6 +66,7 @@ func (s TxAttemptState) String() (str string) {
 }
 
 type TxRequest[ADDR types.Hashable, TX_HASH types.Hashable] struct {
+	RequestID        *TX_HASH
```

Comment:
Could you change this to string?
We just require clients to send us a unique string.
TX_HASH has a certain meaning, implying his field is kind of a TxHash. We don't actually care that this field be same as the TX_HASH type.

So this will be a string here, and also in the DB then.